pkg/openstack/common.go

```diff
@@ -7,12 +7,32 @@ import (
 
 	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
 	"github.com/openstack-k8s-operators/lib-common/modules/common/util"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
 	apimeta "k8s.io/apimachinery/pkg/api/meta"
 	"k8s.io/apimachinery/pkg/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// CheckDeleteDisabled - Delete the object which in turn will clean the sub resources
+func CheckDeleteDisabled(ctx context.Context, helper *helper.Helper, obj client.Object) (ctrl.Result, error) {
+	key := client.ObjectKeyFromObject(obj)
+	if err := helper.GetClient().Get(ctx, key, obj); err != nil {
+		if k8s_errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+	// Delete the object
+	if obj.GetDeletionTimestamp().IsZero() {
```

Comment:
Why we need the timestamp check here? Is it here to avoid queueing up multiple delete requests for an object that is already being deleted?

Reply:
Yes

Reply:
Thanks. It probably won't hurt. :)